client/src/http-helpers.js

```diff
@@ -12,15 +14,23 @@ const getAddress = (id, callback) => {
 
 const postType = (id, type, callback) => {
   $.post('/', { type, id })
-    .done(data => callback(null, data));
+    .done(() => callback())
+    .fail(err => err);
 };
 
-const getRestaurant = (id, callback) => {
-  getTitle(id, (err, data) => {
-    getAddress(id, (error, result) => (
-      error ? callback(error, null) : callback(null, [data[0], data[1], result[0]])
-    ));
-  });
-};
+// unexpected block statement surrounding arrow body
+const getRestaurant = (id, callback) => (
+  new Promise((resolve, reject) => {
+    getTitle(id, (err, data) => {
+      if (err) {
+        reject(err);
+      }
+      resolve(data);
+    });
+  }).then((data) => {
+    getAddress(id, (error, res) => (
+      error ? callback(error, null) : callback(null, [data[0], data[1], res[0]])));
```

Comment:
Is this bad practice to use a callback here? I mixed this in mainly so I could still test this function. Would it be better to use promises only and not mix promises with callbacks?